src/unix/fs.c

```diff
@@ -217,6 +217,21 @@ static ssize_t uv__fs_mkdtemp(uv_fs_t* req) {
 }
 
 
+static ssize_t uv__fs_mkstemp(uv_fs_t* req) {
+  static int no_cloexec_support;
+
+#ifdef O_CLOEXEC
+  no_cloexec_support = 1;
+#endif
+
+  if (no_cloexec_support == 1) {
```

Comment:
> glibc added mkostemp in version 2.7, but we support 2.5

Correct on both accounts.  I didn't look at the pull request closely, but one solution is to look up mkostemp() or mkostemps() through dlsym(), with fallback code for when they're not available.